### ISSUE TITLE
Show Tab close button ("x") on hover

### ISSUE THREAD

**Issue by piotrzarzycki21** [prominic/Moonshine-IDE#124]
Currently in order to close some Tab we need to select it. As part of this feature user will be able to close Tab without selecting it. Close button should be shown on hover.

**Comment by piotrzarzycki21** [comment]
Add comments please if you make some thoughts on that one.

**Comment by JoelProminic** [comment]
I confirmed that the close button appears on hover, and that the button works without changing tab focus.

I also tried closing a file with changes using the same process.  This gave the correct options, and the options worked fine.  However, the "See file again" button did not change focus to the tab.  This makes sense if we think of this button as a "Cancel", but the behavior does not match the text of the button.  We could resolve this by changing the button to "Cancel", or by having the button change focus to the selected tab - I don't have a strong preference.
